### PR TITLE
Add tests for System.Runtime.Serialization.Primitives

### PR DESCRIPTION
**Pull request by hughbe**

### CODE REVIEW

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs]: ```// The .NET Foundation licenses this file to you under the MIT license.```
is missing from the license. please add it.

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/ContractNamespaceAttributeTests.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/ContractNamespaceAttributeTests.cs]: same here. Please add the line to all files.

File: src/System.Runtime/tests/System/Runtime/Serialization/StreamingContextTests.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        public void Ctor_StreamingContextStates_Object(StreamingContextStates state, object additional)
+        {
+            var context = new StreamingContext(state, additional);
+            Assert.Equal(state, context.State);
+            Assert.Equal(additional, context.Context);
+        }
+
+        public static IEnumerable<object[]> Equals_TestData()
+        {
+            yield return new object[] { new StreamingContext(StreamingContextStates.All, null), new StreamingContext(StreamingContextStates.All, null), true };
```
Comment (Anipik) [inline src/System.Runtime/tests/System/Runtime/Serialization/StreamingContextTests.cs]: Cab we use ```TheoryData`` format for these tests.

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        [InlineData(false)]
+        public void IsReference_Set_GetReturnsExpected(bool value)
+        {
+            var attribute = new CollectionDataContractAttribute() { IsReference = value };
+            Assert.Equal(value, attribute.IsReference);
+            Assert.True(attribute.IsReferenceSetExplicitly);
+        }
+
+        [Theory]
+        [InlineData(null)]
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs]: can combine all the InlineData for all the tests into separate ```theoryData<string>``` property

File: src/System.Runtime/tests/System/Runtime/Serialization/OnDeserializedAttributeTests.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.Serialization.Tests
+{
+    public class OnDeserializedAttributeTests
+    {
+        [Fact]
+        public void Ctor_Default()
```
Comment (Anipik) [inline src/System.Runtime/tests/System/Runtime/Serialization/OnDeserializedAttributeTests.cs]: whts the purpose oh this test ?

File: src/System.Runtime/tests/System/Runtime/Serialization/OnSerializingAttributeTests.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+using Xunit;
+
+namespace System.Runtime.Serialization.Tests
+{
+    public class OnSerializingAttributeTests
+    {
+        [Fact]
+        public void Ctor_Default()
+        {
+            new OnSerializingAttribute();
```
Comment (Anipik) [inline src/System.Runtime/tests/System/Runtime/Serialization/OnSerializingAttributeTests.cs]: Do u plan to add more tests to these classes ? if not, combine all the ctor tests in one file

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/IgnoreDataMemberAttributeTests.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Runtime.Serialization.Tests
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/IgnoreDataMemberAttributeTests.cs]: change namespace to ```System.Runtime.Serialization.Primitives.Tests

File: src/System.Runtime.Serialization.Primitives/tests/System.Runtime.Serialization.Primitives.Tests.csproj [hunk 8]
```diff
@@ -0,0 +1,17 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <ProjectGuid>{43341081-27A8-4CFB-8B89-C63CE457EAC2}</ProjectGuid>
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System.Runtime.Serialization.Primitives.Tests.csproj]: this line is not required as no pointers are used here

File: src/System.Runtime.Serialization.Primitives/tests/Configurations.props [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      netstandard;
+    </BuildConfigurations>
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/Configurations.props]: why the tests are just built for netstandard ? why not netcoreapp, netfx also ?

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/IgnoreDataMemberAttributeTests.cs (hunk 7 above)
Comment (hughbe) [reply src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/IgnoreDataMemberAttributeTests.cs]: The convention is to namespace the class for the type its testing so this is already correct

File: src/System.Runtime.Serialization.Primitives/tests/System.Runtime.Serialization.Primitives.Tests.csproj (hunk 8 above)
Comment (hughbe) [reply src/System.Runtime.Serialization.Primitives/tests/System.Runtime.Serialization.Primitives.Tests.csproj]: Fixed

File: src/System.Runtime.Serialization.Primitives/tests/Configurations.props (hunk 9 above)
Comment (hughbe) [reply src/System.Runtime.Serialization.Primitives/tests/Configurations.props]: My understanding is you go for the most broad configuration (i.e. if there is nothing netfx/netcore specific you use netstandard) as the build tools with automatically pick up the right binary if you pass `/P:TargetGroup=netfx`

File: src/System.Runtime/tests/System/Runtime/Serialization/OnDeserializedAttributeTests.cs (hunk 5 above)
Comment (hughbe) [reply src/System.Runtime/tests/System/Runtime/Serialization/OnDeserializedAttributeTests.cs]: I'm gonna remove these tests and they don't pull any weight - I wanted to make sure they do not throw

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs (hunk 4 above)
Comment (hughbe) [reply src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs]: Fixed

File: src/System.Runtime/tests/System/Runtime/Serialization/StreamingContextTests.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Runtime/tests/System/Runtime/Serialization/StreamingContextTests.cs]: Fixed

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/CollectionDataContractAttributeTests.cs]: Fixed for all cases

File: src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/DataContractAttributeTests.cs [hunk 10]
```diff
@@ -1,5 +1,6 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // See the LICENSE file in the project root for more information.
+// The .NET Foundation licenses this file to you under the MIT license.
```
Comment (Anipik) [inline src/System.Runtime.Serialization.Primitives/tests/System/Runtime/Serialization/DataContractAttributeTests.cs]: The order of the lines is different.
```
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
```
Can you please correct it ?

**Comment by hughbe** [comment]
Fixed :)

**Comment by Anipik** [comment]
@danmosemsft can you review this a look too ? It looks good to me

**Comment by danmosemsft** [comment]
Thanks !

**Pull request merged**